### PR TITLE
bump common version to 23.2.1, maps version to 10.10.0 and nav native version to 123.0.0

### PR DESCRIPTION
**Pull request by Zayankovsky**

### CODE REVIEW

**Review by LukasPaczos — COMMENTED** [review]
@Zayankovsky we should be able to revert more of the changes like https://github.com/mapbox/mapbox-navigation-android/pull/6659/files#diff-b28e19257f62571292f230f9103803b19551a0749579af573325bee171d88d21R46, could you give that a pass and ensure that we can go back to the previous definitions without any compilation errors?

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,8 @@ Mapbox welcomes participation and contributions from everyone.
 - Fixed `BannerInstructions` issue where the banner instruction might have been removed from `RouteProgress` at some point around a edge's leg. [#6684](https://github.com/mapbox/mapbox-navigation-android/pull/6684)
 - Added `MapboxNavigation#resetTripSession(callback)` and deprecated the counterpart without a callback. [#6685](https://github.com/mapbox/mapbox-navigation-android/pull/6685)
 - Fixed memory leak in `ReplayProgressObserver` which happened after route refresh. [#6691](https://github.com/mapbox/mapbox-navigation-android/pull/6691)
+- Changed EH speedLimit behaviour: value is picked based on the date-time conditions using device’s system time and edge's location based timezone. [#6702](https://github.com/mapbox/mapbox-navigation-android/pull/6702)
```
[the remark this answers is not in the document]
Comment (Zayankovsky) [reply CHANGELOG.md:13]: But we didn't have this in 2.10.0-beta.2, so I figured it should be in the changelog for 2.10.0-beta.3, shouldn't it?

[the remark this answers is not in the document]
Comment (LukasPaczos) [reply CHANGELOG.md:14]: I don't think so. Anything fixed/backported is always available in later releases as well, otherwise, we'd be regressing.

**Review by LukasPaczos — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (LukasPaczos) [reply CHANGELOG.md:14]: We can address during the chaneglog review though.

**Pull request merged**